compiler/src/dotty/tools/dotc/config/CommandLineParser.scala

```diff
@@ -39,13 +40,18 @@ object CommandLineParser:
       while !terminal do bump()
       !done
 
-    @tailrec def skipToDelim(): Boolean =
+    // Skip to a word boundary, where words can be quoted and quotes can be escaped; false on missing quote.
+    def skipToDelim(): Boolean =
+      var escaped = false
       inline def quote() = { qpos += pos ; bump() }
-      cur match
+      def advance(): Boolean = cur match
```

Comment:
could advance be a separate loop called before the wider pattern match? then skipToDelim will continue to be tailrec.